### PR TITLE
Add message activity/application

### PR DESCRIPTION
**Pull request by MinnDevelopment**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

### Changes

- [x] Internal code
- [x] Library interface (affecting end-user code) 
- [ ] Documentation
- [ ] Other: \_____ 

Closes Issue: NaN

## Description

This is a backport of a JDA v3 feature.

### CODE REVIEW

File: src/main/java/net/dv8tion/jda/api/entities/MessageActivity.java [hunk 1]
```diff
@@ -0,0 +1,223 @@
+/*
+ *     Copyright 2015-2018 Austin Keener & Michael Ritter & Florian Spieß
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/api/entities/MessageActivity.java]: This needs to be updated

File: src/main/java/net/dv8tion/jda/internal/entities/EntityBuilder.java [hunk 2]
```diff
@@ -803,6 +803,35 @@ public Message createMessage(JSONObject jsonObject, MessageChannel chan, boolean
         final List<MessageEmbed>       embeds      = map(jsonObject, "embeds",      this::createMessageEmbed);
         final List<MessageReaction>    reactions   = map(jsonObject, "reactions",   (obj) -> createMessageReaction(chan, id, obj));
 
+        MessageActivity activity = null;
+
+        if (!jsonObject.isNull("activity"))
+        {
+            JSONObject activityData = jsonObject.getJSONObject("activity");
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/internal/entities/EntityBuilder.java]: This should be moved to a method

File: src/main/java/net/dv8tion/jda/internal/entities/EntityBuilder.java [hunk 3]
```diff
@@ -844,15 +849,44 @@ public Message createMessage(JSONObject jsonObject, MessageChannel chan, boolean
...
+
+            application = new MessageActivity.Application(name, description, iconId, coverId, applicationId);
+        }
+        if (activityType == MessageActivity.ActivityType.UNKNOWN)
+        {
+            LOG.debug("Received an unknown ActivityType, Activity: {}", activityData);
         }
 
+        activity = new MessageActivity(activityType, partyId, application);
+        return activity;
```
Comment (DV8FromTheWorld) [inline src/main/java/net/dv8tion/jda/internal/entities/EntityBuilder.java]: `return new MessageActivity(activityType, partyId, application);`

File: src/main/java/net/dv8tion/jda/internal/entities/EntityBuilder.java [hunk 4]
```diff
@@ -884,6 +860,35 @@ public Message createMessage(JSONObject jsonObject, MessageChannel chan, boolean
...
+        final String partyId = activityData.optString("party_id", null);
+        MessageActivity.Application application = null;
+
+        if (!jsonObject.isNull("application"))
+        {
+            JSONObject applicationData = jsonObject.getJSONObject("application");
+
+            final String name = applicationData.getString("name");
+            final String description = applicationData.getString("description");
+            final String iconId = applicationData.getString("icon");
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/internal/entities/EntityBuilder.java]: Apparently this is nullable too

**Pull request merged**